### PR TITLE
libcurl: lowercase system_libs for mingw@Linux + re-format options

### LINKED ISSUE #3976: [package] libcurl/7.66.0 WinSSL not enabled with CMake build
I'm trying to build libcurl/7.66.0 with WINSSL option enabled for my project. But option is disabled in CMakeCache.txt after building is done.
Recipe libcurl/7.66.0 (Latest Revision be0504c6acf1da62622a18855ce40cb2) trying to set only CMAKE_USE_SCHANNEL while configuring CMake. And it does not set CMAKE_USE_WINSSL

It doesn't matter if I use Recipe's option `:with_winssl=True` or `:with_ssl=schannel`. The first one only results in warning:
 _libcurl/7.66.0: WARN: with_openssl, with_winssl, darwin_ssl and with_wolfssl options are deprecated. Use with_ssl option instead._

libcurl-7.66 does not have an option CMAKE_USE_SCHANNEL in CMakeLists.txt. The option was added in later versions. 
There is only option CMAKE_USE_WINSSL and  variables: SSL_ENABLED, USE_SCHANNEL, USE_WINDOWS_SSPI depending on it.
```
if(CMAKE_USE_WINSSL)
  set(SSL_ENABLED ON)
  set(USE_SCHANNEL ON) # Windows native SSL/TLS support
  set(USE_WINDOWS_SSPI ON) # CMAKE_USE_WINSSL implies CURL_WINDOWS_SSPI
  list(APPEND CURL_LIBS "crypt32")
endif()
```

Generated CMakeCache.txt:
```
//No help, variable specified on the command line.
CMAKE_USE_SCHANNEL:UNINITIALIZED=True

//enable Windows native SSL/TLS
CMAKE_USE_WINSSL:BOOL=OFF
```

**Environment details:**
Operation system: Windows 10
Conan version 1.31.2

Profile msvc16StaticDebug64:
```
[settings]
os=Windows
os_build=Windows
arch=x86_64
arch_build=x86_64
compiler=Visual Studio
compiler.version=16
compiler.runtime=MTd
build_type=Debug
[options]
[build_requires]
[env]

```
My conanfile.txt:
```
[requires]
libcurl/7.66.0

[generators]
cmake_find_package

[options]
libcurl:with_ssl=schannel
```

**Step to reproduce:**
conan install --profile msvc16StaticDebug64 --build=missing ..\conanfile.txt

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **libcurl/all**

Fixes #3976

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [X] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by SSE4 — APPROVED** [review]

File: recipes/libcurl/all/conanfile.py [hunk 1]
```diff
@@ -17,46 +17,48 @@ class LibcurlConan(ConanFile):
...
-                       "with_libpsl": False,
-                       "with_largemaxwritesize": False,
-                       "with_nghttp2": False,
-                       "with_zlib": True,
-                       "with_brotli": False,
-                       "with_zstd": False,
-                       "with_c_ares": False,
-                       }
+    options = {
+        "shared": [True, False],
```
Comment (Croydon) [inline recipes/libcurl/all/conanfile.py:21]:
```suggestion
       "shared": [True, False],
```

If we already re-formatting then let's be nit-picking here, this line has a space more than the rest

Comment (madebr) [reply recipes/libcurl/all/conanfile.py:21]: Actually, that was the only line with the intended indentation (=multiple of 4).
Thanks for the thorough review :stuck_out_tongue:

Comment (Croydon) [reply recipes/libcurl/all/conanfile.py:21]: Haha, sorry, correct indentation is hard to see in the web UI

**Review by Croydon — DISMISSED** [review]

File: recipes/libcurl/all/conanfile.py [hunk 2]
```diff
@@ -484,7 +472,10 @@ def _configure_cmake(self):
         self._cmake.definitions["BUILD_SHARED_LIBS"] = self.options.shared
         self._cmake.definitions["CURL_STATICLIB"] = not self.options.shared
         self._cmake.definitions["CMAKE_DEBUG_POSTFIX"] = ""
-        self._cmake.definitions["CMAKE_USE_SCHANNEL"] = self.options.with_ssl == "schannel"
+        if tools.Version(self.version) >= "7.72.0":
+            self._cmake.definitions["CMAKE_USE_SCHANNEL"] = self.options.with_ssl == "schannel"
+        else:
+            self._cmake.definitions["CMAKE_USE_WINSSL"] = self.options.with_ssl == "schannel"
```
Comment (SpaceIm) [inline recipes/libcurl/all/conanfile.py:478]: AFAIR, `CMAKE_USE_SCHANNEL` works in 7.64.1 (oldest version in CCI).

Comment (SpaceIm) [reply recipes/libcurl/all/conanfile.py:478]: mmhh, no actually you're right.

**Review by SpaceIm — APPROVED** [review]

**Comment by SpaceIm** [comment]
Could you also bump zstd please?

**Review by SpaceIm — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Pull request merged**